docs/notebooks/flax_basics.ipynb

```diff
@@ -0,0 +1,911 @@
+{
```

Comment:
<p>Why using random dummy input here? Makes the cell more complicated that it need to be...</p><p> </p><p>Why don't we call them simply <code>params</code>  ?</p><p> </p><p>Having <code>init_params</code>  and alter <code>params</code>  makes it very easy to forget to <code>del init_params</code>  then for large models we have a lot of RAM wasted for no good reason.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google/flax/pull/542/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Reply:
<p>Random versus <code>jnp.array(something)</code>you  mean ? I'm not sure it's that complicated though, but the point was to be able to change the size easily and then see that the shape inference does its job. But maybe that's not the best way to show it.</p><p> </p><p>Re: <code>init_params</code>vs <code>params</code> , I don't have any opinion about it, I just reused the same pattern than the other notebooks, but I don't mind keeping params. Although when I read it I thought it was emphasising the fact that those are only the initial one (and are going to be mutated).</p>

Reply:
If you only care about shape then you should use `jnp.ones()` or `jnp.zeros()`.

I would only use random where we really need random (e.g. for initializing weight) and then taking care to properly derive all random seeds from an initial seed.

@avital WDYT about `init_params`? I noticed this pattern in a couple of places, but for the reason I mentioned in my comment, I think it would be preferable to always call params `params`. But maybe there was a good reason to call them `init_params` the first time?